integration-testing/src/main/java/io/grpc/testing/integration/AbstractTransportTest.java

```diff
@@ -225,6 +225,52 @@ public void clientStreaming() throws Exception {
     assertEquals(goldenResponse, responseObserver.firstValue().get());
   }
 
+  /**
+   * Client sends requests only after receiving a message from server.
+   *
+   * <p>Tests that the header frame can be sent out separately to initiates a new stream
+   * on the server side.
+   */
+  @Test(timeout = 10000)
+  public void checkedUpload() throws Exception {
+    // Makes an emptyUnary call first, to make sure the Settings ACK is sent, so the headers
+    // frame will be sent separately.
+    emptyUnary();
+    final List<StreamingInputCallRequest> requests = Arrays.asList(
+        StreamingInputCallRequest.newBuilder()
+            .setPayload(Payload.newBuilder()
+                .setBody(ByteString.copyFrom(new byte[27182])))
+            .build(),
+        StreamingInputCallRequest.newBuilder()
+            .setPayload(Payload.newBuilder()
+                .setBody(ByteString.copyFrom(new byte[8])))
+            .build(),
+        StreamingInputCallRequest.newBuilder()
+            .setPayload(Payload.newBuilder()
+                .setBody(ByteString.copyFrom(new byte[1828])))
+            .build(),
+        StreamingInputCallRequest.newBuilder()
+            .setPayload(Payload.newBuilder()
+                .setBody(ByteString.copyFrom(new byte[45904])))
+            .build());
+    final StreamingInputCallResponse goldenResponse = StreamingInputCallResponse.newBuilder()
+        .setAggregatedPayloadSize(74922)
+        .build();
+
+    StreamRecorder<StreamingInputCallResponse> responseObserver = StreamRecorder.create();
+    StreamObserver<StreamingInputCallRequest> requestObserver =
+        asyncStub.checkedUploadCall(responseObserver);
+    StreamingInputCallResponse response = responseObserver.firstValue().get();
+    // Server sent a special message to tell us start uploading.
+    assertEquals(-1, response.getAggregatedPayloadSize());
+    for (StreamingInputCallRequest request : requests) {
```

Comment:
No point, just copied the code from the clientStreaming test.

I'll change it to just 1.